module/modmanager/manager.go

```diff
@@ -136,26 +182,63 @@ func (mgr *Manager) Handles() map[string]modlib.HandlerMap {
 
 	res := map[string]modlib.HandlerMap{}
 
-	for n, m := range mgr.modules {
+	mgr.modules.Range(func(n string, m *module) bool {
 		res[n] = m.handles
-	}
+		return true
+	})
 
 	return res
 }
 
-// Add adds and starts a new resource module.
-func (mgr *Manager) Add(ctx context.Context, conf config.Module) error {
-	mgr.mu.Lock()
-	defer mgr.mu.Unlock()
-	return mgr.add(ctx, conf)
-}
-
-func (mgr *Manager) add(ctx context.Context, conf config.Module) error {
+// Add adds and starts a new resource modules for each given module configuration.
+//
+// Each module configuration should have a unique name - if duplicate names are detected,
+// then only the first duplicate instance will be processed and the rest will be ignored.
+//
+// This method is not thread-safe.
```

Comment:
Awesome! Super clear

Reply:
thanks again for the de-duping suggestion